discovery-infra/test_infra/utils.py

```diff
@@ -771,9 +766,33 @@ def copy_template_tree(dst, none_platform_mode=False):
     )
 
 
+@contextmanager
+def pull_secret_file():
+    pull_secret = os.environ.get("PULL_SECRET")
+
+    try:
+        json.loads(pull_secret)
+    except json.JSONDecodeError as e:
+        raise ValueError(f"Value of PULL_SECRET environment variable "
+                         f"is not a valid JSON payload") from e
+
+    pull_secret_path = None
+    try:
+        with tempfile.NamedTemporaryFile(mode='w', delete=False) as f:
+            f.write(pull_secret)
+            pull_secret_path = f.name
+
+        yield pull_secret_path
+
+    finally:
+        if pull_secret_path and os.path.exists(pull_secret_path):
+            os.unlink(pull_secret_path)
```

Comment:
can't the yield be inside the `with` and then the `delete=True` would be called as expected? Then, you don't need to explicitly call to unlink.